src/dectris/neggia/data/PathResolverV2.cpp

```diff
@@ -0,0 +1,114 @@
+#include "PathResolverV2.h"
+#include <assert.h>
+#include <memory>
+#include "H5BTreeVersion2.h"
+#include "H5FractalHeap.h"
+#include "H5LocalHeap.h"
+#include "constants.h"
+
+#include <iostream>
+
+PathResolverV2::PathResolverV2(const H5ObjectHeader& root) : _root(root) {}
+
+ResolvedPath PathResolverV2::resolve(const H5Path& path) {
+    return resolvePathInHeader(_root, path);
+}
+
+ResolvedPath PathResolverV2::resolvePathInHeader(const H5ObjectHeader& in,
+                                                 const H5Path& path) {
+    H5ObjectHeader parentEntry(path.isAbsolute() ? _root : in);
+    for (auto itemIterator = path.begin(); itemIterator != path.end();
+         ++itemIterator)
+    {
+        auto item = *itemIterator;
+        try {
+            auto resolvedPath = findPathInObjectHeader(
+                    parentEntry, item, H5Path(path, itemIterator + 1));
+            if (resolvedPath.externalFile.get() != nullptr) {
+                // the object is in a different file which must be opened by
+                // upper layer
+                return resolvedPath;
+            }
+            if ((itemIterator + 1) == path.end()) {
+                return resolvedPath;
+            }
+            parentEntry = resolvedPath.objectHeader;
+        } catch (const std::out_of_range&) {
```

Comment:
So what is actually going on here? If the path can't be found the whole path is returned?